File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/MailSchemaExtraInfoTables.cpp


#include "MailSchema_extraInfoTables.h"

using namespace Tiny;

MailSchema_extraInfoTables::MailSchema_extraInfoTables()
{
	mail = MailExtraInfoTable();
	tutorials = MailTutorialsTable();
}

MailSchema_extraInfoTables::MailSchema_extraInfoTables(std::string jsonString)
{
	this->fromJson(jsonString);
}

MailSchema_extraInfoTables::~MailSchema_extraInfoTables()
{

}

void
MailSchema_extraInfoTables::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *mailKey = "mail";

    if(object.has_key(mailKey))
    {
        bourne::json value = object[mailKey];




        MailExtraInfoTable* obj = &mail;
		obj->fromJson(value.dump());

    }

    const char *tutorialsKey = "tutorials";

    if(object.has_key(tutorialsKey))
    {
        bourne::json value = object[tutorialsKey];




        MailTutorialsTable* obj = &tutorials;
		obj->fromJson(value.dump());

    }


}

bourne::json
MailSchema_extraInfoTables::toJson()
{
    bourne::json object = bourne::json::object();






	object["mail"] = getMail().toJson();






	object["tutorials"] = getTutorials().toJson();


    return object;

}

MailExtraInfoTable
MailSchema_extraInfoTables::getMail()
{
	return mail;
}

void
MailSchema_extraInfoTables::setMail(MailExtraInfoTable mail)
{
	this->mail = mail;
}

MailTutorialsTable
MailSchema_extraInfoTables::getTutorials()
{
	return tutorials;
}

void
MailSchema_extraInfoTables::setTutorials(MailTutorialsTable tutorials)
{
	this->tutorials = tutorials;
}

File: OPENAPI-REST-API/swagger-client/spring/src/main/java/io/swagger/model/MailSchemaExtraInfoTables.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.model.MailExtraInfoTable;
import io.swagger.model.MailTutorialsTable;
import io.swagger.v3.oas.annotations.media.Schema;
import org.springframework.validation.annotation.Validated;
import org.openapitools.jackson.nullable.JsonNullable;
import io.swagger.configuration.NotUndefined;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Additional information tables for the mail service.
 */
@Schema(description = "Additional information tables for the mail service.")
@Validated
@NotUndefined



public class MailSchemaExtraInfoTables   {
  @JsonProperty("mail")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private MailExtraInfoTable mail = null;

  @JsonProperty("tutorials")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private MailTutorialsTable tutorials = null;


  public MailSchemaExtraInfoTables mail(MailExtraInfoTable mail) { 

    this.mail = mail;
    return this;
  }

  /**
   * Get mail
   * @return mail
   **/
  
  @Schema(description = "")
  
@Valid
  public MailExtraInfoTable getMail() {  
    return mail;
  }



  public void setMail(MailExtraInfoTable mail) { 
    this.mail = mail;
  }

  public MailSchemaExtraInfoTables tutorials(MailTutorialsTable tutorials) { 

    this.tutorials = tutorials;
    return this;
  }

  /**
   * Get tutorials
   * @return tutorials
   **/
  
  @Schema(description = "")
  
@Valid
  public MailTutorialsTable getTutorials() {  
    return tutorials;
  }



  public void setTutorials(MailTutorialsTable tutorials) { 
    this.tutorials = tutorials;
  }

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    MailSchemaExtraInfoTables mailSchemaExtraInfoTables = (MailSchemaExtraInfoTables) o;
    return Objects.equals(this.mail, mailSchemaExtraInfoTables.mail) &&
        Objects.equals(this.tutorials, mailSchemaExtraInfoTables.tutorials);
  }

  @Override
  public int hashCode() {
    return Objects.hash(mail, tutorials);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class MailSchemaExtraInfoTables {\n");
    
    sb.append("    mail: ").append(toIndentedString(mail)).append("\n");
    sb.append("    tutorials: ").append(toIndentedString(tutorials)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
